node.go

```diff
@@ -522,8 +523,8 @@ func leafNodesAreEqual(node1, node2 Node) bool {
 }
 
 func groupNodesAreEqual(node1, node2 Node) bool {
-	fields1 := node1.Fields()
-	fields2 := node2.Fields()
+	fields1 := slices.Clone(node1.Fields())
+	fields2 := slices.Clone(node2.Fields())
```

Comment:
Let's move the close after the length check so we only pay for it after the cheap check was passed?